braidpool-primitives/src/bead/mod.rs

```diff
@@ -64,6 +69,62 @@ impl Bead {
         // TODO: Implement this function.
         unimplemented!()
     }
+
+    #[inline]
+    pub fn is_parent_of(&self, child_bead_hash: BeadHash) -> bool {
+        self.children.borrow().contains(&child_bead_hash)
+    }
+
+    pub fn is_genesis_bead(&self, braid: &Braid) -> bool {
+        if self.parents.is_empty() {
+            return true;
+        };
+
+        // We need to check whether even one of the parent beads have been pruned from memory!
+        for (parent_bead_hash, _) in self.parents.iter() {
+            let parent_bead = braid.load_bead_from_hash(*parent_bead_hash);
+            if let Err(error_type) = parent_bead {
+                match error_type {
+                    BeadLoadError::BeadPruned => return true,
+                    _ => panic!("Fatal Error Detected!"),
+                };
+            }
+        }
+
+        false
+    }
+
+    #[inline]
+    pub fn is_orphaned(&self, braid: &Braid) -> bool {
+        for (parent, _) in self.parents.iter() {
+            if braid.beads.contains(parent) == false {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    #[inline]
+    pub fn get_parents(&self) -> Parents {
```

Comment:
Why would we need a `get_parents()` function instead of just using `bead.parents` (assuming bead is a public struct).